content/spin/plugin-authoring.md

```diff
@@ -0,0 +1,41 @@
+title = "Creating Spin plugins"
+template = "spin_main"
+date = "2023-01-26T00:22:56Z"
+[extra]
+url = "https://github.com/fermyon/spin/blob/main/docs/content/plugin-authoring.md"
+---
+
+Spin plugins add new functionality or subcommands to Spin without modifying the Spin codebase. They make Spin easily extensible while keeping it lightweight. Spin plugins can add new triggers to Spin (such as the [example timer trigger](https://github.com/fermyon/spin/blob/main/examples/spin-timer/trigger-timer.json)), enable new language SDKs (such as [`js2wasm`](https://github.com/fermyon/spin-plugins/blob/main/manifests/js2wasm/js2wasm.json)), and more.
+
+This document will cover what Spin plugins are, how to use a plugin, and how to contribute a plugin.
+
+## What are Spin plugins?
+
+A Spin plugin is an executable that is added to Spin's plugins directory (`$XDG_DATA_HOME/spin/plugins`) upon a `spin plugins install <plugin-name>`. The plugin is then ready to be used. If the plugin is an extension to the Spin CLI, it can now be executed directly as a subcommand: `spin <plugin-name>`. If the plugin is a trigger plugin, it will be executed during `spin up` when an app using that trigger is run. 
+
+While for now plugins are assumed to be executables, in the future, support for plugging in WebAssembly modules may be desirable.
+
+## How to find and use a Spin plugin
+
+Spin maintains a [Spin plugins repository](https://github.com/fermyon/spin-plugins) that acts as an inventory of available plugins. The `spin plugins` command can be used to list plugins (available and installed) and install a new plugin. For example, the `js2wasm` plugin is needed in order to write JavaScript Spin applications. It can be installed from the Spin plugins repository, using the following command: 
+
+```sh
+spin plugins install js2wasm
+```
+
+During `spin build`, `spin js2wasm` is called to build the JS Spin application. Learn more about building Spin components in JavaScript [here](https://developer.fermyon.com/spin/javascript-components.md).
```

Comment:
This made me think that the plugin somehow altered the behaviour of `spin build`.  Maybe something like "With the plugin installed, you can call `spin js2wasm` as part of building a JavaScript component - for example, in the component build directive."?

Reply:
Good point! I struggled a little with this part. Technically `npm build` is the build directive to JS Spin apps which aliases [to a command](https://github.com/fermyon/spin-js-sdk/blob/3a069cb45de54b627ca1e523882aaf244310e178/examples/javascript/hello_world/package.json#L7) that contains `spin js2wasm`.  Maybe we say, as part of the npm build directive. @karthik2804 any ideas here?

Reply:
Maybe: "With the plugin installed, you can now call `spin js2wasm` to run it.  In this case, for example, you might call it from your JavaScript application's `npm build` script."  (That we call the npm script from `spin build` is incidental to what I think you're trying to highlight here, namely the ability to write `spin js2wasm` - please correct me if I've misunderstood the intent.)